pkg/apis/apps/v1alpha1/staticpod_types.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// StaticPodUpgradeStrategy defines a strategy to upgrade a static pod.
+type StaticPodUpgradeStrategy struct {
+	// Type of Static Pod upgrade. Can be "auto" or "ota".
+	Type StaticPodUpgradeStrategyType `json:"type,omitempty"`
+
+	// Auto upgrade config params. Present only if type = "auto".
+	//+optional
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+}
+
+// StaticPodUpgradeStrategyType is a strategy according to which a static pod gets upgraded.
+type StaticPodUpgradeStrategyType string
+
+const (
+	AutoStaticPodUpgradeStrategyType StaticPodUpgradeStrategyType = "auto"
+	OTAStaticPodUpgradeStrategyType  StaticPodUpgradeStrategyType = "ota"
+)
+
+// StaticPodSpec defines the desired state of StaticPod
+type StaticPodSpec struct {
+	// StaticPodName indicates the static pod desired to be upgraded.
+	StaticPodName string `json:"staticPodName"`
+
+	// StaticPodManifest indicates the Static Pod desired to be upgraded. The corresponding
+	// manifest file name is `StaticPodManifest.yaml`.
+	// +optional
+	StaticPodManifest string `json:"staticPodManifest,omitempty"`
+
+	// Namespace indicates the namespace of target static pod
+	// +optional
+	StaticPodNamespace string `json:"staticPodNamespace,omitempty"`
+
+	// An upgrade strategy to replace existing static pods with new ones.
+	UpgradeStrategy StaticPodUpgradeStrategy `json:"upgradeStrategy,omitempty"`
+
```

Comment:
how about add a field `RevisionHistoryLimit *int32` for recording `The number of old history to retain to allow rollback`. Defaults to 10.